src/testRunner/unittests/tsserver/updateFileSystem.ts

```diff
@@ -0,0 +1,125 @@
+            const requests: [string, Partial<protocol.Request>][] = [
+                ["Initial updateFileSystem", {
+                    command: protocol.CommandTypes.UpdateFileSystem,
+                    arguments:{
+                        fileSystem: "memfs",
+                        files: [app, file1, file2, file3, config, lib],
+                        deleted: [],
+                    }
+                }],
+                ["Opening app.ts", {
+                    command: protocol.CommandTypes.Open,
+                    arguments: { file: app.file }
+                }],
+                ["Opening file3.ts", {
+                    command: protocol.CommandTypes.Open,
+                    arguments: {
+                        file: file3.file,
+                        fileContent: fileContentWithComment(file3)
+                    }
+                }],
+                ["non-delete", {
+                    command: protocol.CommandTypes.UpdateFileSystem,
+                    arguments:{
+                        fileSystem: "memfs",
+                        files: [],
+                        deleted: [],
+                    }
+                }],
+                ["delete", {
+                    command: protocol.CommandTypes.UpdateFileSystem,
+                    arguments:{
+                        fileSystem: "memfs",
+                        files: [],
+                        deleted: [file1.file],
+                    }
+                }],
+                ["close", {
+                    command: protocol.CommandTypes.Close,
+                    arguments: { file: app.file }
+                }],
+            ];
+            VirtualFS.createWatcher(fshost.fsWatches, "/host/b/app.ts" as Path, {
+                cb: (_, relativeFileName) => {
+                    assert.fail("The watcher for /host/b/app.ts should not be called at all. Called with " + relativeFileName);
+                },
+                directoryName: "/host/b",
+                fallbackOptions: {},
+                fallbackPollingInterval: PollingInterval.Medium,
+            });
+
+            const scenario = "updateFileSystem";
+            const subScenario = "open and close files";
+            baselineFileSystem(scenario, subScenario, requests, fshost, session);
+
+            host.ensureFileOrFolder({
+                path: "/host/b/app.ts",
+                content: 'console.log("hello")'
+            });
+            host.modifyFile("/host/b/app.ts", "console.log('goodbye')");
+            assert.isFalse(host.fileExists("/fshost/b/app.ts"), "host fileExists /fshost/b/app.ts");
+            assert.isTrue(fshost.fileExists("/fshost/b/app.ts"), "fshost fileExists /fshost/b/app.ts");
+            assert.isTrue(host.fileExists("/host/b/app.ts"), "host fileExists /host/b/app.ts");
+            assert.isFalse(fshost.fileExists("/host/b/app.ts"), "fshost fileExists /host/b/app.ts");

```

Comment:
Can you instead add this to baseline (that is diff fsHost and host after ensureFileOrFolder to show only host is changed)